### DISCUSSION TITLE
Issue Passing TestRail_CLI environment variables to YAML file

### DISCUSSION THREAD

**Discussion by nikampak** [gurock/trcli#213] [Q&A] [answered]
Can you help me to find a way to pass my test rail cli environment variables to YAML file.
I have followed this link [ https://support.testrail.com/hc/en-us/articles/12974525736084-Parameter-sources#configuration-files-0-1:~:text=as%20a%20parameter.-,Environment%20variables,-It%20is%20possible ] to set up the environment variables in .zshrc file as following:
   export TR_CLI_USERNAME=username@company.com
   export TR_CLI_PASSWORD=password

From the idea, I changed the trcli-config.yml file to following:

   host: https://company.testrail.io/
   project: ProjectName
   username: ${TR_CLI_USERNAME}
   password: ${TR_CLI_PASSWORD}
   file: absolutePath.xml
   config: absolutePath/trcli-config.yml
   title: titleOfTestRun
   milestone_id: milestoneId
   run_id: testRunId

Then I run command:

   trcli -y -c "absolutePathToFile/trcli-config.yml" parse_junit --case-matcher "name"

But have an issue saying that Error detected while checking a project: 'Authentication failed: invalid or missing user/password or session cookie.'

**Comment by nikampak** [comment]
PS: my environment - gradle, junit5, kotlin, appium

**Reply by bitcoder** [reply, marked as answer]
Hi @nikampak ,
you can set the variables in your shell. You don't have to, and shouldn't,  put any reference to them on the .yml configuration file.
